src/deepsparse/engine.py

```diff
@@ -170,22 +200,39 @@ def __init__(
         self._cpu_avx_type = AVX_TYPE
         self._cpu_vnni = VNNI
 
+        if context is not None and (scheduler is not None and scheduler != "elastic"):
+            raise ValueError(
+                "Contexts are currently only supported when running the elastic scheduler"
+            )
+        if context is not None and context.num_cores() != num_cores:
+            raise ValueError(
+                "The Context was initialized with {} cores and the Engine was initialized"
+                " with {}. These two values should be equal.".format(
+                    str(context.num_cores()), str(num_cores)
+                )
+            )
+
+        model_path = self._model_path
         if self._input_shapes:
-            with override_onnx_input_shapes(
+            model_path = override_onnx_input_shapes(
```

Comment:
we need to keep the `with` statement here so the temporary file generated by `override_onnx_input_shapes` will be cleaned up after the engine is created